lms/djangoapps/grades/tests/test_tasks.py

```diff
@@ -59,24 +59,79 @@ def set_up_course(self, enable_subsection_grades=True):
         _ = anonymous_id_for_user(self.user, self.course.id)
         # pylint: enable=attribute-defined-outside-init,no-member
 
-    def test_score_changed_signal_queues_task(self):
+    @ddt.data(
+        ('lms.djangoapps.grades.tasks.recalculate_subsection_grade.apply_async', SCORE_CHANGED),
+        ('lms.djangoapps.grades.tasks.recalculate_course_grade.apply_async', COURSE_GRADE_UPDATE_REQUESTED)
+    )
+    @ddt.unpack
+    def test_signal_queues_task(self, enqueue_op, test_signal):
         """
-        Ensures that the SCORE_CHANGED signal enqueues a recalculate subsection grade task.
+        Ensures that the SCORE_CHANGED and COURSE_GRADE_UPDATE_REQUESTED signals enqueue the correct tasks.
         """
         self.set_up_course()
+        if test_signal == SCORE_CHANGED:
+            expected_args = self.score_changed_kwargs.values()
+        else:
+            expected_args = [self.score_changed_kwargs['user_id'], self.score_changed_kwargs['course_id']]
         with patch(
-            'lms.djangoapps.grades.tasks.recalculate_subsection_grade.apply_async',
+            enqueue_op,
             return_value=None
         ) as mock_task_apply:
-            SCORE_CHANGED.send(sender=None, **self.score_changed_kwargs)
-            mock_task_apply.assert_called_once_with(args=tuple(self.score_changed_kwargs.values()))
+            test_signal.send(sender=None, **self.score_changed_kwargs)
+            mock_task_apply.assert_called_once_with(args=tuple(expected_args))
 
-    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
-    def test_subsection_grade_updated(self, default_store):
+    @patch('lms.djangoapps.grades.signals.signals.COURSE_GRADE_UPDATE_REQUESTED.send')
+    def test_subsection_update_triggers_course_update(self, mock_course_signal):
+        """
+        Ensures that the subsection update operation also updates the course grade.
+        """
+        self.set_up_course()
+        recalculate_subsection_grade.apply(args=tuple(self.score_changed_kwargs.values()))
+        mock_course_signal.assert_called_once_with(
+            sender=recalculate_subsection_grade,
+            course_id=self.score_changed_kwargs['course_id'],
+            user_id=self.score_changed_kwargs['user_id'],
+        )
+
+    @ddt.data(True, False)
+    def test_course_update_enqueuing(self, should_be_async):
+        """
+        Ensures that the course update operation is enqueued on an async queue (or not) as expected.
+        """
+        base = 'lms.djangoapps.grades.tasks.recalculate_course_grade.apply'
+        if should_be_async:
+            execute = base + '_async'
```

Comment:
Nit: Might be clearer with:

    base = 'lms.djangoapps.grades.tasks.recalculate_course_grade'
    async_task = base + '.apply_async'
    inline_task = base + '.apply'